addons/pkg-npm/model-java/src/main/java/org/commonjava/indy/pkg/npm/model/PackageMetadata.java

```diff
@@ -261,9 +277,293 @@ public void setAttachments( Object attachments )
+                }
+                catch ( ParseException e )
+                {
+                    logger.error( String.format( "Cannot parse date when comparing. Reason: %s", e ) );
+                }
+                return compare;
+            } );
+
+            Map<String, String> clone = new LinkedHashMap<>();
+
+            if ( time.get( MODIFIED ) != null )
+            {
+                clone.put( MODIFIED, time.get( MODIFIED ) );
+            }
+            if ( time.get( CREATED ) != null )
+            {
+                clone.put( CREATED, time.get( CREATED ) );
+            }
+
+            for ( final Map.Entry<String, String> entry : timeList )
+            {
+                if ( MODIFIED.equals( entry.getKey() ) || CREATED.equals( entry.getKey() ) )
+                {
+                    continue;
+                }
+                clone.put( entry.getKey(), entry.getValue() );
+            }
+            time = clone;
+        }
+
+        // merge dist-tag object
+        DistTag sourceDist = source.getDistTags();
+        Map<String, String> sourceDistMap = sourceDist.fetchTagsMap();
+        Map<String, String> thisDistMap = distTags.fetchTagsMap();
+
+        if ( sourceDistMap.size() > 0 )
+        {
+            if ( thisDistMap == null || thisDistMap.size() <= 0 )
+            {
+                this.setDistTags( sourceDist );
+                changed = true;
+            }
+            else
+            {
+                for ( final String tag : sourceDistMap.keySet() )
+                {
+                    if ( !thisDistMap.keySet().contains( tag ) )
+                    {
+                        distTags.putTag( tag, sourceDistMap.get( tag ) );
+                        changed = true;
+                    }
+                    else
+                    {
+                        int compare = VersionUtils.createSingleVersion( sourceDistMap.get( tag ) )
+                                                  .compareTo( VersionUtils.createSingleVersion(
+                                                                  thisDistMap.get( tag ) ) );
+                        if ( compare > 0 )
+                        {
+                            distTags.putTag( tag, sourceDistMap.get( tag ) );
+                            changed = true;
+                        }
+                    }
+                }
+            }
+        }
+
+        //merge versions
+        Map<String, VersionMetadata> sourceVersions = source.getVersions();
+        Iterator vm = sourceVersions.keySet().iterator();
+
+        while ( vm.hasNext() )
+        {
+            String v = (String) vm.next();
+            VersionMetadata value = sourceVersions.get( v );
+            boolean s = false;
+            Iterator snapshot = versions.keySet().iterator();
+
+            while ( snapshot.hasNext() )
+            {
+                String preExisting = (String) snapshot.next();
+                if ( preExisting.equals( v ) )
```

Comment:
Shouldn't we keep the original value if they're the same? It seems like that would be more consistent with the null-check-then-set approach you use for simple fields above.